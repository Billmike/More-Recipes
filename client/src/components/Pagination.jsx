import React from 'react';
import ReactPaginate from 'react-paginate';

const Pagination = props => (
  <div>
    <ReactPaginate
      previousLabel="Previous"
      nextLabel="Next"
      breakLabel={<a href="">...</a>}
      breakClassName="page-link"
      onPageChange={props.handlePaginationChange}
      pageCount={props.pageCount}
      containerClassName="pagination pagination-lg custom-pagination"
      pageLinkClassName="page-link"
      nextLinkClassName="page-link"
      previousLinkClassName="page-link"
      disabledClassName="disabled"
      pageClassName="page-item"
      previousClassName="page-item"
      nextClassName="page-item"
      activeClassName="active"
      subContainerClassName="pages pagination"
    />
  </div>
);

export default Pagination;
