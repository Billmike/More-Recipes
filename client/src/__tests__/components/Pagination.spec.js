import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../../components/Pagination';

describe('<Pagination />', () => {
  it('Should render the Pagination component correctly', () => {
    const wrapper = shallow(<Pagination />);
    expect(wrapper).toMatchSnapshot();
  });
});
