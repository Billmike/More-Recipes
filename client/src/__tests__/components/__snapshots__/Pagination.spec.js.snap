// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Should render the Pagination component correctly 1`] = `
<div>
  <PaginationBoxView
    activeClassName="active"
    breakClassName="page-link"
    breakLabel={
      <a
        href=""
      >
        ...
      </a>
    }
    containerClassName="pagination pagination-lg custom-pagination"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={3}
    nextClassName="page-item"
    nextLabel="Next"
    nextLinkClassName="page-link"
    pageClassName="page-item"
    pageCount={10}
    pageLinkClassName="page-link"
    pageRangeDisplayed={2}
    previousClassName="page-item"
    previousLabel="Previous"
    previousLinkClassName="page-link"
    subContainerClassName="pages pagination"
  />
</div>
`;
